share/namespace.go

```diff
@@ -2,21 +2,41 @@ package share
 
 import (
 	"bytes"
+	"encoding/hex"
+	"encoding/json"
 	"fmt"
+	"slices"
 )
 
 type Namespace struct {
 	data []byte
 }
 
+// MarshalJSON encodes namespace to the json encoded bytes.
+func (n Namespace) MarshalJSON() ([]byte, error) {
+	return json.Marshal(n.data)
+}
+
+// UnmarshalJSON decodes json bytes to the namespace.
+func (n *Namespace) UnmarshalJSON(data []byte) error {
+	var buf []byte
+	if err := json.Unmarshal(data, &buf); err != nil {
+		return err
+	}
+
+	ns, err := NewNamespaceFromBytes(buf)
+	if err != nil {
+		return err
+	}
+	*n = ns
+	return nil
+}
+
 // NewNamespace validates the provided version and id and returns a new namespace.
 // This should be used for user specified namespaces.
 func NewNamespace(version uint8, id []byte) (Namespace, error) {
-	if err := ValidateUserNamespace(version, id); err != nil {
-		return Namespace{}, err
-	}
-
-	return newNamespace(version, id), nil
+	ns := newNamespace(version, id)
+	return ns, ns.ValidateUserNamespace()
 }
```

Comment:
Why have you changed this such that the namespace is still returned even if invalid?

Reply:
it will return an error anyway in case it happened. Decided to have fewer lines of code.

Reply:
but it does allow someone to potentially ignore the error and use the invalid namespace

Reply:
Agree with @cmwaters, that returning both invalid value and error is unusual and might be confusing for user

Reply:
done